spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/system/injection/ProviderMethod.java

```diff
@@ -0,0 +1,68 @@
+/*
+ *  Copyright (c) 2020 - 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.spi.system.injection;
+
+import org.eclipse.dataspaceconnector.spi.system.Provider;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import static java.lang.String.format;
+import static java.util.Optional.ofNullable;
+
+/**
+ * Represents a {@link Method} that is annotated with the {@link Provider} annotation.
+ */
+public class ProviderMethod {
+    private final Method method;
+    private final boolean isDefault;
+
+    public ProviderMethod(Method method) {
+        this.method = method;
+        isDefault = ofNullable(method.getAnnotation(Provider.class)).map(Provider::isDefault)
+                .orElseThrow(() -> new IllegalArgumentException(format("Method %s is not annotated with @Provider!", method)));
```

Comment:
I guess it's a matter of preference, but strictly speaking this deals with validating _input_, rather than actual dependency injection, therefore I went with the `IllegalArgumentException`. Theoretically `ProviderMethod`s could be used for something else too